### PR TITLE
More rich output format

### PR DESCRIPTION
**Pull request by drug007**
Trying to merge old changes to upstream. Major changes are additional output of data address and data in hex format. Old output is preserved.
Output before:
```
fixarray: 3 items
  positive fixint: 11
  positive fixint: 8
  fixarray: 5 items
    uint8: 255
    positive fixint: 0
    positive fixint: 7
    uint8: 170
    positive fixint: 85
```
Output after:
```
0x000000: fixarray: 3 items                       ( 0x93 )
0x000001:   positive fixint: 11                            [0x0b]
0x000002:   positive fixint: 8                             [0x08]
0x000003:   fixarray: 5 items                     ( 0x95 )
0x000004:     uint8: 255                          ( 0xcc ) [0xff]
0x000006:     positive fixint: 0                           [0x00]
0x000007:     positive fixint: 7                           [0x07]
0x000008:     uint8: 170                          ( 0xcc ) [0xaa]
0x00000a:     positive fixint: 85                          [0x55]
```
Metadata are in braces, data are in brackets.

### CODE REVIEW

File: src/msgpackdump.d [hunk 1]
```diff
@@ -9,99 +9,104 @@ import std.stdio;
...
-	else return u;
 }
 
 private T read16(T)(ref size_t index, const ubyte[] bytes) if (T.sizeof == 2)
 {
 	auto buf = bswap(*(cast(ushort*) (bytes.ptr + index)));
-        T r = *(cast(T*) &buf);
-        index += 2;
+		T r = *(cast(T*) &buf);
+		index += 2;
```
Comment (Hackerpilot) [inline src/msgpackdump.d]: These lines and the corresponding lines in `read32` and `read64` should only be indented one level.

File: src/msgpackdump.d [hunk 2]
```diff
@@ -127,229 +132,323 @@ void main(string[] args)
...
+			// but copy there only data chunk that is placeable
+			// to the buffer
+			formatted = cast(char[])format(fmt, args)[0..TextWidth - pos];
+			copy(formatted, line[pos..TextWidth]);
+		}
+		pos += formatted.length;
+		assert (pos < TextWidth);
+	}
+
+	void refactorMeformattedPrintToLine(Char, Args...)(in Char[] fmt, Args args)
```
Comment (Hackerpilot) [inline src/msgpackdump.d]: Is the "Meformatted" part a typo? Why is "refactor" part of the name?

File: src/msgpackdump.d (hunk 1 above)
Comment (drug007) [reply src/msgpackdump.d]: I see

File: src/msgpackdump.d (hunk 2 above)
Comment (drug007) [reply src/msgpackdump.d]: Oh, this is the artifact I missed. I'll fix it.

**Comment by drug007** [comment]
I was forced to refactor all. That increases the PR size but in total I've made the code cleaner and simpler I think.

**Review by Hackerpilot — APPROVED** [review]

**Pull request merged**